gtsfm/scene_optimizer.py

```diff
@@ -274,13 +277,27 @@ def _filter_dict_keys(dict: Dict[Any, Any], ref_dict: Dict[Any, Any]) -> Dict[An
         if self._save_gtsfm_data:
             auxiliary_graph_list.extend(save_gtsfm_data(image_graph, ba_input_graph, ba_output_graph))
 
+        img_dict_graph = dask.delayed(get_image_dictionary)(image_graph)
+        dense_points_graph = self.dense_multiview_optimizer.create_computation_graph(img_dict_graph, ba_output_graph)
+        # Cast to string as Open3d cannot use PosixPath's for I/O -- only string file paths are accepted.
+        auxiliary_graph_list.append(
+            dask.delayed(io_utils.save_point_cloud_as_ply)(save_fpath=str(MVS_PLY_SAVE_FPATH), points=dense_points_graph)
+        )
+
         # as visualization tasks are not to be provided to the user, we create a
         # dummy computation of concatenating viz tasks with the output graph,
         # forcing computation of viz tasks
         output_graph = dask.delayed(lambda x, y: (x, y))(ba_output_graph, auxiliary_graph_list)
+        ba_output_graph = output_graph[0]
 
         # return the entry with just the sfm result
-        return output_graph[0]
+        return ba_output_graph
+
+
+def get_image_dictionary(image_list: List[Image]) -> Dict[int, Image]:
+    """Convert a list of images to the MVS input format."""
+    img_dict = {i: img for i, img in enumerate(image_list)}
+    return img_dict
```

Comment:
Why do we need a function for one line of code? 

If its required for some reason, maybe we should move it into the MVS module instead of in the SceneOptimizer?